xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp

```diff
@@ -255,30 +267,41 @@ void CLogindUPowerSyscall::InhibitDelayLock()
 
   if (!reply)
   {
-    CLog::Log(LOGWARNING, "LogindUPowerSyscall - failed to inhibit sleep delay lock");
-    m_delayLockFd = -1;
-    return;
+    CLog::Log(LOGWARNING, "LogindUPowerSyscall - failed to inhibit %s delay lock", what);
+    return -1;
   }
 
-  if (!dbus_message_get_args(reply, NULL, DBUS_TYPE_UNIX_FD, &m_delayLockFd, DBUS_TYPE_INVALID))
+  int delayLockFd;
+  if (!dbus_message_get_args(reply, NULL, DBUS_TYPE_UNIX_FD, &delayLockFd, DBUS_TYPE_INVALID))
   {
     CLog::Log(LOGWARNING, "LogindUPowerSyscall - failed to get inhibit file descriptor");
-    m_delayLockFd = -1;
-    return;
+    return -1;
   }
 
-    CLog::Log(LOGDEBUG, "LogindUPowerSyscall - inhibit lock taken, fd %i", m_delayLockFd);
+  CLog::Log(LOGDEBUG, "LogindUPowerSyscall - inhibit lock taken, fd %i", delayLockFd);
+  return delayLockFd;
 #else
-    CLog::Log(LOGWARNING, "LogindUPowerSyscall - inhibit lock support not compiled in");
+  CLog::Log(LOGWARNING, "LogindUPowerSyscall - inhibit lock support not compiled in");
+  return -1;
 #endif
 }
 
-void CLogindUPowerSyscall::ReleaseDelayLock()
+void CLogindUPowerSyscall::ReleaseDelayLockSleep()
+{
+  ReleaseDelayLock(m_delayLockSleepFd, "sleep");
+}
+
+void CLogindUPowerSyscall::ReleaseDelayLockShutdown()
+{
+  ReleaseDelayLock(m_delayLockShutdownFd, "shutdown");
+}
+
+void CLogindUPowerSyscall::ReleaseDelayLock(int& lockFd, const char *what)
```

Comment:
I know we have no rules on this, so consider it a personal request: Don't change the value of non-`const` references that were given as parameters. It looks like a normal method invocation and hides the fact that the argument value itself will be changed, making the code difficult to follow. I'd suggest just writing the few lines out. Or you could even use `CFileHandle`.

Reply:
I made a small change and moved the reset to -1 outside of the ReleaseDelayLock method.
Please take a look if you like it better that way. Thx.

Reply:
Great, thanks!